primitives/externalities/src/extensions.rs

```diff
@@ -87,6 +88,10 @@ pub trait ExtensionStore {
 	/// It is advised to use [`ExternalitiesExt::extension`](crate::ExternalitiesExt::extension)
 	/// instead of this function to get type system support and automatic type downcasting.
 	fn extension_by_type_id(&mut self, type_id: TypeId) -> Option<&mut dyn Any>;
+
+	fn register_extension_with_type_id(&mut self, type_id: TypeId, extension: Box<dyn Extension>) -> Result<(), Error>;
```

Comment:
Can you suggest the change that allows to avoid it?

Reply:
That would be using a `VerificationExt` the same way you got a `TaskExecutorExt`, (init will be in StateMachine::new).
The content of `VerificationExt` will be optional. So it can be activated/instantiated on `start_verify` in the same way it is currently activated by registering an new extension.
This means, as you mention, that in some case a `VerificationExt` will be accessible to context where we do not want it but it will not do much as it will be `None`.
So I think the new register functions allow to implement this kind of mechanism in a more generic way, that can be fine, I am just not sure if it fits the design for `Extension` (@tomusdrw or someone else more familiar with extension design could say).

Reply:
Oh, dynamic extensions (which you can register and deregister, and do not need to bookkeep every one of them in `StateMachine`) is one of the stretch goals of the PR (see branch name :) )